### PR TITLE
[IMP] portal, website: move lang picker to portal

### PR DESCRIPTION
**Pull request by fja-odoo**
Make the language switcher available on portal without website installed

task-2203383

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/portal/models/ir_ui_view.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            to render portal layout template)
+        """
+        qcontext = super(View, self)._prepare_qcontext()
+        if request and getattr(request, 'is_frontend', False):
+            qcontext.update(dict(
+                self._context.copy(),
+                languages=request.env['res.lang'].get_available(),
+                url_for=url_for,
+            ))
+        return qcontext
```
Comment (qsm-odoo) [inline addons/portal/models/ir_ui_view.py:25]: @JKE-be should not this be in http_routing?

File: addons/portal/models/res_lang.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class Lang(models.Model):
+    _inherit = "res.lang"
+
+    @api.model
+    def get_available(self):
+        """ Return the available languages as a list of (code, name) sorted by name. """
+        if request and request.is_frontend_multilang:
+            langs = self.search([])
+            return sorted([(lang.code, lang.url_code, lang.name) for lang in langs], key=itemgetter(2))
+        return super(Lang, self).get_available()
```
Comment (qsm-odoo) [inline addons/portal/models/res_lang.py:18]: I don't know about this without investigating... @JKE-be ?

File: addons/website/views/website_templates.xml [hunk 3]
```diff
@@ -545,33 +542,16 @@
...
-                        <a t-att-href="url_for(request.httprequest.path + '?' + keep_query(), lang_code=lg[0])"
-                           class="dropdown-item js_change_lang"
-                           t-att-data-url_code="lg[1]">
-                            <t t-esc="lg[2].split('/').pop()"/>
-                        </a>
-                    </t>
-                </div>
-            </div>
-        </li>
+<template id="add_language" name="Website Language Selector" inherit_id="portal.language_selector">
```
Comment (qsm-odoo) [inline addons/website/views/website_templates.xml:545]: Apparently renamed just to annoy me... :)

File: addons/portal/models/ir_ui_view.py (hunk 1 above)
Comment (qsm-odoo) [reply addons/portal/models/ir_ui_view.py:25]: decided that no :+1:

File: addons/portal/models/res_lang.py (hunk 2 above)
Comment (qsm-odoo) [reply addons/portal/models/res_lang.py:18]: TODO
- replace is_frontend_multilang in is_frontend
- add a comment for the active_test removal (or better)

**Comment by qsm-odoo** [comment]
Continuing on https://github.com/odoo/odoo/pull/55300

**Comment by fja-odoo** [comment]
@qsm-odoo Sorry, je suis un peu short niveau temps en ce moment... cette tache commencait à arriver en haut de la liste !

**Comment by qsm-odoo** [comment]
@fja-odoo Je ne te demandais rien anyway ;)

**Pull request closed without merging**